plugins/hypervisors/xenserver/src/main/java/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1441,7 +1441,8 @@ public VM createVmFromTemplate(final Connection conn, final VirtualMachineTO vmS
             throw new CloudRuntimeException("Unable to finalize VM MetaData: " + vmSpec);
         }
         try {
-            setVmBootDetails(vm, conn, vmSpec.getBootType(), vmSpec.getBootMode());
+            String bootMode = org.apache.commons.lang3.StringUtils.defaultIfEmpty(vmSpec.getDetails().get(ApiConstants.BootType.UEFI.toString()), null);
+            setVmBootDetails(vm, conn, bootMode == null ? null : ApiConstants.BootType.UEFI.toString(), bootMode);
```

Comment:
@shwstppr Why boot mode is hard coded to UEFI if not specified ? Is it possible to hard-code based on any specific version of xcp-ng?

Reply:
@sureshanaparti setVmBootDetails() is taking both bootType and bootMode params. I've changed the code a bit for more readability. Please check now.